packages/js-drive/lib/abci/handlers/prepareProposalHandlerFactory.js

```diff
@@ -0,0 +1,49 @@
+const {
+  tendermint: {
+    abci: {
+      ResponsePrepareProposal,
+    },
+  },
+} = require('@dashevo/abci/types');
+
+const txAction = {
+  UNKNOWN: 0, // Unknown action
+  UNMODIFIED: 1, // The Application did not modify this transaction.
+  ADDED: 2, // The Application added this transaction.
+  REMOVED: 3, // The Application wants this transaction removed from the proposal and the mempool.
+};
+
+/**
+ *
+ * @return {prepareProposalHandler}
+ */
+function prepareProposalHandlerFactory() {
+  /**
+   * @typedef prepareProposalHandler
+   * @param {abci.RequestPrepareProposal} request
+   * @return {Promise<abci.ResponsePrepareProposal>}
+   */
+  async function prepareProposalHandler({
+    maxTxBytes,
+    txs,
+  }) {
+    let totalSizeBytes = 0;
+
+    const txRecords = txs.filter((tx) => {
```

Comment:
Does it make sense to iterate over all transactions when we already know that max size is reached?